### DISCUSSION TITLE
Is it supported to use on the simulator? or only use on real device?

### DISCUSSION THREAD

**Discussion by youngfreeFJS** [danielpaulus/go-ios#260] [Q&A]
Is it supported to use on the simulator? or only use on real device?

**Comment by wellkilo** [comment]
go-ios targets **physical iOS devices**, not CoreSimulator devices.

Its transport and services are built around usbmuxd/lockdown (and RemoteXPC tunnels on iOS 17+), pairing records, developer images and device services. An iOS Simulator is managed by Apple's CoreSimulator framework and simctl instead, and it does not appear as a go-ios usbmux device.

On macOS, use Xcode's command-line tool for simulator automation, for example:

    xcrun simctl list devices
    xcrun simctl boot 'iPhone 16'
    xcrun simctl install booted MyApp.app
    xcrun simctl launch booted com.example.app

For UI tests, run them through xcodebuild against a simulator destination. On Linux/Windows there is no supported Apple iOS Simulator runtime, so go-ios cannot provide one.

The project README describes its device-facing architecture and features (pairing devices, installing developer images, device services) rather than CoreSimulator support: [README](https://github.com/danielpaulus/go-ios/blob/c584042907dee35bf80377284ae3aac85990b4d7/README.md#features).
